### ISSUE TITLE
Re-design zarr checksum to not rely on "full path"s (and become a proper tree checksum)

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandi-archive#931]

**Comment by dchiquito** [comment]
@yarikoptic you are correct about the current design, it does use full paths in the checksum files. I did it that way so that you could identify the exact object described in the .checksum file entry, but if you were able to obtain the .checksum file then you already have the containing directory, so the information is technically redundant.

I don't have an objection to removing that information, but it would require a recalculation of every existing zarr checksum, or some kind of versioned hash scheme. The current scheme is `dandi:dandi-zarr-checksum`. We would need to either alter the definition of that scheme or introduce a new one. @AlmightyYakob if we altered the checksum definition as Yarik proposes, the recalculation script could be used to redo all the checksums, correct?

**Comment by AlmightyYakob** [comment]
> @AlmightyYakob if we altered the checksum definition as Yarik proposes, the recalculation script could be used to redo all the checksums, correct?

Yes, we could just run it on all affected dandisets/zarrs.

**Comment by yarikoptic** [comment]
since we do not have yet (much) of dandisets with zarrs in main deployment, may be indeed we could take a shortcut and just redefine/recompute `dandi:dandi-zarr-checksum` instead of coming up with a new (e.g., `dandi:dandi-zarr-checksum2` ;)).  we would just need to rush/coordinate change of compute in dandi-cli as well, and declare prior state of the dandiverse (both dandischema, dandi-cli?) as "bad", right?

**Comment by dchiquito** [comment]
I'm addressing this in https://github.com/dandi/dandi-archive/issues/937
